scripts/collection_from_items.py

```diff
@@ -115,28 +118,37 @@ def main() -> None:
 
     s3_client = client("s3")
 
-    files_to_read = list_json_in_uri(uri, s3_client)
+    files_to_read = list_files_in_uri(uri, [SUFFIX_JSON, SUFFIX_FOOTPRINT], s3_client)
 
     start_time = time_in_ms()
+    polygons = []
     for key, result in get_object_parallel_multithreading(
         bucket_name_from_path(uri), files_to_read, s3_client, arguments.concurrency
     ):
-        item_stac = json.loads(result["Body"].read().decode("utf-8"))
-
-        if not arguments.collection_id == item_stac.get("collection"):
-            get_log().trace(
-                "skipping: item.collection != collection.id",
-                file=key,
-                action="collection_from_items",
-                reason="skip",
-            )
-            continue
-
-        collection.add_item(item_stac)
-        get_log().info("item added to collection", item=item_stac["id"], file=key)
+        content = json.load(result["Body"])
+        # The following if/else looks like it could be avoid by refactoring `list_files_in_uri()`
+        # to return a result list per suffix, but we would have to call `get_object_parallel_multithreading()`
+        # for each of them to avoid this if/else.
+        if key.endswith(SUFFIX_JSON):
+            if not arguments.collection_id == content.get("collection"):
```

Comment:
`not` is usually used for things like `x not in y`; personally I find it clearer when there's a single comparator rather than a comparator and a negation:

```suggestion
            if arguments.collection_id != content.get("collection"):
```